robot.py

```diff
@@ -1,166 +1,177 @@
-        # are changes to self.<containerName>
-        container = getattr(self, containerName)
-
-        subsystems = config.getSubsystems()
-        createdCount = 0
-        for subsystem in subsystems:
-            items = {key: factory(descp) for (key, descp) in config.getGroupDict(subsystem, groupName).items()}
-            if(len(items) == 0):
-                continue
-            container[subsystem] = items
-            createdCount += len(container[subsystem])
-            groupName_subsystem = "_".join([groupName,subsystem])
-            self.logger.info("Creating %s", groupName_subsystem)
-            setattr(self, groupName_subsystem, container[subsystem])
-
-        self.logger.info(f"Created {createdCount} items for {groupName} groups with `{factory.__name__}` into `{containerName}")
-
-
-if __name__ == '__main__':
-    wpilib.run(MyRobot)
+"""
+Team 3200 Robot base class
+"""
+# Module imports:
+import wpilib
+from wpilib import XboxController
+from wpilib import SerialPort
+from magicbot import MagicRobot, tunable
+
+# Component imports:
+from components.driveTrain import DriveTrain
+from components.pneumatics import Pneumatics
+from components.buttonManager import ButtonManager, ButtonEvent
+from components.breakSensors import Sensors
+from components.winch import Winch
+from components.shooterMotors import ShooterMotorCreation
+from components.shooterLogic import ShooterLogic
+from components.loaderLogic import LoaderLogic
+from components.elevator import Elevator
+from components.scorpionLoader import ScorpionLoader
+from components.feederMap import FeederMap
+from components.lidar import Lidar
+
+# Other imports:
+from robotMap import RobotMap, XboxMap
+from utils.componentUtils import testComponentCompatibility
+from utils.motorHelper import createMotor
+from utils.sensorFactories import gyroFactory, breaksensorFactory
+from utils.acturatorFactories import compressorFactory, solenoidFactory
+import utils.math
+
+# Test imports:
+from components.testBoard import TestBoard
+
+
+class MyRobot(MagicRobot):
+    """
+    Base robot class of Magic Bot Type
+    """
+    shooter: ShooterLogic
+    loader: LoaderLogic
+    feeder: FeederMap
+    sensors: Sensors
+    shooterMotors: ShooterMotorCreation
+    driveTrain: DriveTrain
+    winch: Winch
+    buttonManager: ButtonManager
+    pneumatics: Pneumatics
+    elevator: Elevator
+    scorpionLoader: ScorpionLoader
+    lidar: Lidar
+
+    # Test code:
+    testBoard: TestBoard
+
+    sensitivityExponent = tunable(1.8)
+
+    def createObjects(self):
+        """
+        Robot-wide initialization code should go here. Replaces robotInit
+        """
+        self.map = RobotMap()
+        self.xboxMap = XboxMap(XboxController(1), XboxController(0))
+
+        self.MXPserial = SerialPort(115200, SerialPort.Port.kMXP, 8,
+        SerialPort.Parity.kParity_None, SerialPort.StopBits.kStopBits_One)
+        self.MXPserial.setReadBufferSize(18)
```

Comment:
Hardcoded value. Make it a variable so we can eventually change in config or at least document it and make it easily changed.